From f7e43ebdb45cbe681b88a531575ec45d2c253ae1 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Tue, 12 Jan 2021 16:05:02 -0800
Subject: [PATCH 3/3] glx: Don't link libmesautil.la for apple

libmesautil is already included in libmesa, and the linker complains of duplicate symbols

Regressed-in: f4a9d205d8a6474defa86dc4f12b7e63e61a49d9
Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 src/glx/Makefile.am | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/src/glx/Makefile.am b/src/glx/Makefile.am
index 773a5397da5..4aaa73923d7 100644
--- a/src/glx/Makefile.am
+++ b/src/glx/Makefile.am
@@ -99,7 +99,6 @@ libglx_la_SOURCES = \
 
 libglx_la_LIBADD = \
 	$(top_builddir)/src/loader/libloader.la \
-	$(top_builddir)/src/util/libmesautil.la \
 	$(top_builddir)/src/util/libxmlconfig.la
 
 if HAVE_DRISW
@@ -143,10 +142,22 @@ libglx_la_SOURCES += \
 	applegl_glx.c
 
 SUBDIRS += apple
+# libmesa is for _mesa_init_remap_table
 libglx_la_LIBADD += \
 	$(builddir)/apple/libappleglx.la \
 	$(top_builddir)/src/mesa/libmesa.la
+
+# libmesautil.la is already included by libmesa
+libglx_la_LIBADD -= \
+	$(top_builddir)/src/util/libmesautil.la
+
 nodist_EXTRA_lib@GL_LIB@_la_SOURCES = dummy.cpp
+
+else
+
+libglx_la_LIBADD += \
+        $(top_builddir)/src/util/libmesautil.la
+
 endif
 
 if HAVE_WINDOWSDRI
-- 
2.24.3 (Apple Git-128)

